src/Provider/TenantProvider.php

```
@@ -0,0 +1,62 @@
+<?php
+
+namespace RL\Provider;
+
+use RL\Repository\ApiKeyRepository;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class TenantProvider implements UserProviderInterface
+{
+    /** @var ApiKeyRepository */
+    private ApiKeyRepository $apiKeyRepository;
+
+    public function __construct(
+        ApiKeyRepository $apiKeyRepository
+    ) {
+        $this->apiKeyRepository = $apiKeyRepository;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function loadUserByUsername($username)
+    {
+        //pass
```

Maybe we should throw an exception here